app.rb

```diff
@@ -0,0 +1,106 @@
+require './teacher'
+require './student'
+require './book'
+require './rental'
+require './classroom'
+require './person'
+
+class App
+  attr_accessor :books, :students, :teachers, :rentals
+
+  def initialize
+    @books = []
+    @peoples = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each_with_index do |bk, i|
+      puts "#{i}) Title: #{bk.title}, Author: #{bk.author}"
+    end
+  end
+
+  def list_peoples
+    @peoples.each_with_index do |p, i|
+      puts "#{i}) [#{p.class}] Name: #{p.name}, ID: #{p.id}"
+    end
+  end
+
+  def create_student
+    puts 'Lets create a student!'
+    print 'Student Name: '
+    name = gets.chomp
+    print 'Student Age: '
+    age = gets.chomp.to_i
+    print 'Student Classroom: '
+    cl = gets.chomp
+    print 'Do you have parent permission? (true(T) or False(F): '
+    permission = gets.chomp.strip.upcase
+    case permission
+    when 'T'
+      @peoples.push(Student.new(age, name, true, cl))
+      puts 'You have created a Student successfully!'
+    when 'F'
+      @peoples.push(Student.new(age, name, false, cl))
+      puts 'You have created a Student successfully!'
+    else
+      puts 'Invalid'
+    end
+  end
+
+  def create_teacher
+    puts 'Lets create a teacher.'
+    print 'Teacher name: '
+    name = gets.chomp
+    print 'Teacher Age: '
+    age = gets.chomp
+    print 'Teacher\'s specialization: '
+    specialization = gets.chomp
+    @peoples.push(Teacher.new(specialization, age, name))
+    puts 'Teacher created successfully!!'
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    choice = gets.chomp.to_i
+    case choice
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'INVALID NUMBER!!'
+    end
+  end
+
+  def create_book
+    print 'Book Title: '
+    title = gets.chomp
+    print 'Book Author: '
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+    puts 'Book created successfully!'
+  end
+
+  def create_rental
+    puts 'Select a book from the list by number'
+    list_books
+    bk_choice = gets.chomp.to_i
+    puts 'Select a person from the list by number (not id)'
+    list_peoples
+    p_choice = gets.chomp.to_i
+    print 'Enter today\'s date: '
+    date = gets.chomp
+    @rentals.push(Rental.new(date, @books[bk_choice], @peoples[p_choice]))
+    puts 'Rental added successfully'
+  end
```

Comment:
- [ ] I was getting an error when I try to create a rental, let me know if there is anything I was not doing properly. See the image below.

![rental](https://user-images.githubusercontent.com/53124548/200105894-3aeb032f-4e90-4af7-8b44-1ae0bef61d96.png)

Reply:
@elisha2kyakpo1 Thanks for the review. There is nothing with the rental. You need to add a book and a person and then choose the id for each of them first before creating a rental and it will be added successfully.

Reply:
@Elijahdre thanks the comment, but I did create a person and a book, to use them for a rental.

Reply:
@elisha2kyakpo1 please check these
![ruby1](https://user-images.githubusercontent.com/47558964/200109313-cd85056c-1ccd-49ee-a660-0d60eee19bdb.PNG)
![ruby2](https://user-images.githubusercontent.com/47558964/200109323-d4ac0fbe-f7ed-4d3f-be72-849f28e9b139.PNG)
![ruby3](https://user-images.githubusercontent.com/47558964/200109329-8c1950db-6c3d-4ed3-8cfb-b3cf354b74fc.PNG)

Reply:
Great @Elijahdre, the next reviewer will take note